server/src/main/java/org/apache/druid/server/http/MetadataResource.java

```diff
@@ -148,9 +151,25 @@ public Response getDatabaseSegmentDataSource(@PathParam("dataSourceName") final
   @GET
   @Path("/segments")
   @Produces(MediaType.APPLICATION_JSON)
-  public Response getDatabaseSegments(@Context final HttpServletRequest req)
+  public Response getDatabaseSegments(
+      @Context final HttpServletRequest req,
+      @QueryParam("datasources") String datasources
```

Comment:
This can be simply a list or set.

Reply:
Didn't find any existing API's taking in a list as query param in druid code, so went with json string. Also the URL with list will look like `/druid/coordinator/v1/metadata/segments?datasources="foo"&datasources="bar"&datasources="baz"` instead of `/druid/coordinator/v1/metadata/segments?datasources=["foo","bar","baz"]`. But I can change if there's any benefit.